src/main/scala/backgammon/Hash.scala

```diff
@@ -0,0 +1,894 @@
+package strategygames.backgammon
+
+import strategygames.Player
+
+final class Hash(size: Int) {
+
+  def apply(situation: Situation): PositionHash = {
+    val l = Hash.get(situation, Hash.polyglotTable)
+    if (size <= 8) {
+      Array.tabulate(size)(i => (l >>> ((7 - i) * 8)).toByte)
+    } else {
+      val m = Hash.get(situation, Hash.randomTable)
+      Array.tabulate(size)(i =>
+        if (i < 8) (l >>> ((7 - i) * 8)).toByte
+        else (m >>> ((15 - i) * 8)).toByte
+      )
+    }
+  }
+}
+
+object Hash {
+
+  val size = 3
+
+  class ZobristConstants(start: Int) {
+    def hexToLong(s: String): Long =
+      (java.lang.Long.parseLong(s.substring(start, start + 8), 16) << 32) |
+        java.lang.Long.parseLong(s.substring(start + 8, start + 16), 16)
+    val p1TurnMask                 = hexToLong(ZobristTables.p1TurnMask)
+    val actorMasks                 = ZobristTables.actorMasks.map(hexToLong)
+    val pocketMasks                = ZobristTables.pocketMasks.map(hexToLong)
+  }
+
+  object ZobristConstants {}
+
+  // The following masks are compatible with the Polyglot
+  // opening book format.
+  private val polyglotTable    = new ZobristConstants(0)
+  private lazy val randomTable = new ZobristConstants(16)
+
+  // for backgammon hashInt is just 1 for stone
+  private def pieceIndex(piece: Piece, count: Int) =
+    (piece.role.hashInt * count) * 2 + piece.player.fold(1, 0)
+
+  private def actorIndex(actorCount: (Actor, Int)) =
+    Pos.all.size * pieceIndex(actorCount._1.piece, actorCount._2) + actorCount._1.pos.hashCode
+
+  def get(situation: Situation, table: ZobristConstants): Long = {
+
+    def pocketMask(playershift: Int, count: Int) = {
+      if (0 < count && count <= 15)
+        Option(table.pocketMasks(count + playershift))
+      else None
+    }
+
+    val board = situation.board
+    val hturn = situation.player.fold(table.p1TurnMask, 0L)
+
+    val hactors = board.actors.values.view
+      .map {
+        table.actorMasks compose actorIndex _
+      }
+      .fold(hturn)(_ ^ _)
+
+    board.pocketData.fold(hactors) { data =>
```

Comment:
I don't think we need to handle the pocket data as captured pieces always return the same player, therefore the mask from the board and its pieces is sufficient.

Reply:
I think we do need to know, as a position is different depending on if there are pieces in the pocket or not. We only need to track one of pocket and score though as the other is implied by the rest of the board state